inc/zoo/swar/SWAR.h

```diff
@@ -256,16 +256,29 @@ struct BooleanSWAR: SWAR<NBits, T> {
     /// A logical NOT in this circumstance _only_ flips the MSB of each lane.  This operation is
     /// not ones or twos complement.
     constexpr BooleanSWAR operator not() const noexcept {
-        return maskLaneMSB ^ *this;
+        return BooleanSWAR<NBits, T>{maskLaneMSB ^ this->value()};
     }
  private:
     constexpr BooleanSWAR(SWAR<NBits, T> initializer) noexcept: SWAR<NBits, T>(initializer) {}
 
     template<int NB, typename TT> friend
     constexpr BooleanSWAR<NB, TT> greaterEqual_MSB_off(SWAR<NB, TT> left, SWAR<NB, TT> right) noexcept;
+    template<int NB, typename TT> friend
+    constexpr BooleanSWAR<NB, TT> differents(SWAR<NB, TT> left, SWAR<NB, TT> right) noexcept;
 
 };
 
+// Differents becuase "differences of lanes in boolean bits"
+template<int NB, typename TT> constexpr
+BooleanSWAR<NB, TT> differents(SWAR<NB, TT> left, SWAR<NB, TT> right) noexcept {
+    return left ^ right;
+}
```

Comment:
This does not leave the most significant bit set on "true", the MSB must be the boolean, and the rest 0